crates/ruff/src/rules/flake8_django/rules/locals_in_render_function.rs

```diff
@@ -43,11 +45,25 @@ impl Violation for DjangoLocalsInRenderFunction {
 }
 
 /// DJ003
+impl Analyzer<ast::ExprCall> for DjangoLocalsInRenderFunction {
+    fn rule() -> Rule {
+        Rule::DjangoLocalsInRenderFunction
+    }
+
+    fn run(diagnostics: &mut Vec<Diagnostic>, context: &RuleContext, node: &ast::ExprCall) {
+        locals_in_render_function(diagnostics, context, node);
+    }
+}
+
 pub(crate) fn locals_in_render_function(
-    checker: &mut Checker,
-    func: &Expr,
-    args: &[Expr],
-    keywords: &[Keyword],
+    diagnostics: &mut Vec<Diagnostic>,
+    checker: &RuleContext,
+    ast::ExprCall {
+        func,
+        args,
+        keywords,
+        ..
+    }: &ast::ExprCall,
```

Comment:
We would / will need to change every rule to take the relevant AST node, rather than the individual fields-as-arguments, which is a good change IMO (and the thing that prohibited us from doing this earlier).